invertedai/large/common.py

```diff
@@ -0,0 +1,153 @@
+from typing import List, Optional, Tuple
+from pydantic import BaseModel
+
+from invertedai.common import AgentAttributes, AgentState, RecurrentState, Point
+
+class Region(BaseModel):
+    """
+    A data structure to contain information about a region in a map to be used in large simulations.
+
+    See Also
+    --------
+    AgentAttributes
+    """
+
+    center: Optional[Point] = None # The center of the region if such a concept is relevant (e.g. center of a square, center of a rectangle)
+    fov: Optional[float] = None # Side length of the region for the default interpretation of a region as a square
+    agent_states: Optional[List[AgentState]] = None # A list of existing agents within the region
+    agent_attributes: Optional[List[AgentAttributes]] = None # The attributes of agents that exist within the region or that will be initialized within the region
+    recurrent_states: Optional[List[RecurrentState]] = None # Recurrent states of the agents contained within the region
+    region_type: str = 'square' # Geometric category of the region. As of now, only 'square' is supported.
+    vertexes: List[Point] # An ordered list of x-y coordinates of the region defined clockwise around the perimeter of the region
```

Comment:
If you are referring to `vertex`, i believe the plural form is `vertices`.

Reply:
Vertices is indeed used but Vertexes is also used and more grammatically correct.